### PR TITLE
added videoinfer cpp example

### PR DESCRIPTION
**Pull request by dataplayer12**
This PR builds upon the cpp api `infer` utility to handle videos and cameras. A blog explaining these changes is [here](https://towardsdatascience.com/detecting-pedestrians-and-bikers-on-a-drone-with-jetson-xavier-93ce92e2c597).

### CODE REVIEW

**Comment by yashnv** [comment]
@dataplayer12 sorry for getting back late on this. Did you try this in the NGC container? Because this example requires OpenCV to be built with ffmpeg, which is not the default configuration in the Pytorch container. Can you confirm this?

**Comment by yashnv** [comment]
@dataplayer12 did you get a chance to review this in NGC Pytorch 19.09/19.10 container?

**Comment by dataplayer12** [comment]
@yashnv Sorry for the late response. I have not been able to review this on NGC container because my xavier has trouble running docker. Is it not possible to mark this as a jetson only feature since inference is what jetsons are meant for and opencv on jetpack is already compiled with ffmpeg. Otherwise, I could modify the Dockerfile in this repo to additionally compile opencv with ffmpeg after it has pulled the NGC container. Which would you prefer?

**Comment by yashnv** [comment]
@dataplayer12 let's just add this to Jetson(s) for now, changing the environment(compiling opencv) adds complications within containers, which we want to avoid.

File: extras/cppapi/CMakeLists.txt [hunk 1]
```diff
@@ -34,3 +34,6 @@ find_package(OpenCV REQUIRED)
 
 add_executable(infer infer.cpp)
 target_link_libraries(infer PRIVATE retinanet ${OpenCV_LIBS} cuda ${CUDA_LIBRARIES})
+
```
Comment (yashnv) [inline extras/cppapi/CMakeLists.txt]:
```
if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")
  add_executable(infervideo infervideo.cpp)
  target_link_libraries(infervideo PRIVATE retinanet ${OpenCV_LIBS} cuda ${CUDA_LIBRARIES})
endif()
```

**Comment by dataplayer12** [comment]
@yashnv Thanks for making the change. Hope that the PR will be accepted by @james-nvidia

**Pull request merged**